core/src/main/java/org/elasticsearch/indices/cache/query/terms/TermsLookup.java

```diff
@@ -19,29 +19,37 @@
 
 package org.elasticsearch.indices.cache.query.terms;
 
-import org.elasticsearch.common.Nullable;
-import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.query.QueryValidationException;
+
+import java.io.IOException;
+import java.util.Objects;
 
 /**
+ * Encapsulates the parameters needed to fetch terms.
  */
-public class TermsLookup {
+public class TermsLookup implements Writeable<TermsLookup>, ToXContent {
+    static final TermsLookup PROTOTYPE = new TermsLookup();
 
-    private final String index;
-    private final String type;
-    private final String id;
-    private final String routing;
-    private final String path;
+    private String index;
+    private String type;
+    private String id;
+    private String routing;
+    private String path;
 
-    @Nullable
-    private final QueryParseContext queryParseContext;
+    public TermsLookup() {
+    }
 
-    public TermsLookup(String index, String type, String id, String routing, String path, @Nullable QueryParseContext queryParseContext) {
+    public TermsLookup(String index, String type, String id, String routing, String path) {
```

Comment:
I think I would remove this constructor, seems like it's only used in tests and we can replace it with empty constructors + setters